src/components/sbb-radio-button-group/sbb-radio-button-group.tsx

```diff
@@ -199,36 +202,25 @@ export class SbbRadioButtonGroup {
 
   @Listen('keydown')
   public handleKeyDown(evt: KeyboardEvent): void {
-    if (!this._enabledRadios) {
-      return;
-    }
+    const enabledRadios: HTMLSbbRadioButtonElement[] = this._enabledRadios;
 
-    const enabledRadios = this._enabledRadios;
-    const checked = enabledRadios.findIndex((radio) => radio.checked);
-    const cur = checked !== -1 ? checked : 0;
-    const size = enabledRadios.length;
-    const prev = cur === 0 ? size - 1 : cur - 1;
-    const next = cur === size - 1 ? 0 : cur + 1;
-
-    // don't trap nested handling
     if (
-      (evt.target as HTMLElement) !== this._element &&
-      (evt.target as HTMLElement).parentElement !== this._element
+      !enabledRadios ||
+      // don't trap nested handling
+      ((evt.target as HTMLElement) !== this._element &&
+        (evt.target as HTMLElement).parentElement !== this._element)
     ) {
       return;
     }
 
-    const currentWritingMode = getDocumentWritingMode();
-    const prevKey = currentWritingMode === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
-    const nextKey = currentWritingMode === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
-
-    if (evt.key === prevKey || evt.key === 'ArrowUp') {
-      enabledRadios[prev].select();
-      enabledRadios[prev].focus();
-      evt.preventDefault();
-    } else if (evt.key === nextKey || evt.key === 'ArrowDown') {
-      enabledRadios[next].select();
-      enabledRadios[next].focus();
+    if (isArrowKeyPressed(evt)) {
+      const checked: number = enabledRadios.findIndex((radio) => radio.checked);
+      const nextIndex: number = getNextElementIndexFunction(evt)(
+        checked !== -1 ? checked : 0,
+        enabledRadios.length
+      );
```

Comment:
A matter of taste/readability. Take your pick :)
```suggestion
const getNextElementIndex = getNextElementIndexFunction(evt);
      const checked: number = enabledRadios.findIndex((radio) => radio.checked);
      const current: number = checked !== -1 ? checked : 0;
      const nextIndex: number = getNextElementIndex(current, enabledRadios.length);
```